src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -562,6 +566,635 @@ def _encloses_omp_directive(self):
+
+        # Have Reference +/- Literal, analyse
+        # and create clause appropriately
+        index_name = "" 
+        if type(node.children[0]) is Reference:
+            index_name = writer(node.children[0].symbol)
+        if type(node.children[1]) is Reference:
+            index_name = writer(node.children[1].symbol)
+        index_private = (index_name in 
+                         parallel_private)
+        if index_private:
+            if index_name in private_list:
+                # FIXME I think this should be
+                # supportable, but need to think
+                if node.operator is \
+                    BinaryOperation.ADD:
+                    index_name = index_name + ("+({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                else:
+                    index_name = index_name + ("-({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                index_strings.append(index_name)
+            elif index_name in firstprivate_list:
+                # name +/- stop-start
+                if node.operator is \
+                    BinaryOperation.ADD:
+                    index_name = index_name + ("+({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                else:
+                    index_name = index_name + ("-({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                index_strings.append(index_name)
+            else:
+                firstprivate_list.append(index_name)
+                # name +/- stop-start 
+                if node.operator is \
+                    BinaryOperation.ADD:
+                    index_name = index_name + ("+({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                else:
+                    index_name = index_name + ("-({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                index_strings.append(index_name)
+        else:
+            # FIXME I think this should be
+            # supportable, but need to think
+            raise GenerationError(
+                    "Shared variable access used "
+                    "as an index inside an "
+                    "OMPTaskDirective which is not "
+                    "supported.")
+
+    def compute_clauses(self):
+        '''
+        TODO
+        '''
+        from psyclone.psyir.backend.fortran import FortranWriter
+        private_list = []
+        firstprivate_list = []
+        shared_list = []
+        in_list = []
+        out_list = []
+
+        # Create a FortranWriter to convert nodes to strings
+        # TODO: Create issue - support non-Fortran backends
```

Comment:
Yes, I think if you're having to use a FortranWriter to implement functionality then there's something not quite right.